From 6b8a202a36ec1005fbdd1278cb06f268587000dc Mon Sep 17 00:00:00 2001
From: Ciprian Marian Costea <ciprianmarian.costea@nxp.com>
Date: Sat, 30 Jan 2021 10:46:53 +0200
Subject: [PATCH 52/78] s32gen1: fsl-quadspi: Map AHB buffer using
 'ioremap_cache'

As a performance improvement, we try to create a cached
mapping of the AHB buffer, if possible.

Issue: ALB-6395
Signed-off-by: Ciprian Marian Costea <ciprianmarian.costea@nxp.com>
---
 drivers/mtd/spi-nor/fsl-quadspi.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/drivers/mtd/spi-nor/fsl-quadspi.c b/drivers/mtd/spi-nor/fsl-quadspi.c
index aaff3c36b07b..994b2ac7311a 100644
--- a/drivers/mtd/spi-nor/fsl-quadspi.c
+++ b/drivers/mtd/spi-nor/fsl-quadspi.c
@@ -922,7 +922,7 @@ static int fsl_qspi_probe(struct platform_device *pdev)
 
 #ifdef CONFIG_SOC_S32GEN1
 	/* Map AHB Buffer */
-	q->ahb_addr = ioremap(QUADSPI_AHB_BASE, QUADSPI_AHB_SIZE);
+	q->ahb_addr = ioremap_cache(QUADSPI_AHB_BASE, QUADSPI_AHB_SIZE);
 	if (!q->ahb_addr)
 		return -ENOMEM;
 #endif
-- 
2.25.1

